src/Cryptol/TypeCheck/Solve.hs

```diff
@@ -290,10 +290,10 @@ proveImplication dedupErrs lnam as ps gs =
 -- | Tries to prove an implication. If proved, then returns `Right (m_su ::
 -- InferM Subst)` where `m_su` is an `InferM` computation that results in the
 -- solution substitution, and records any warning invoked during proving. If not
--- proved, then returns `Left (m_err :: InferM ())`, which records all errors
+-- proved, then returns `Left (m_err :: [Error])`, which records all errors
 -- invoked during proving.
```

Comment:
Perhaps just:

```suggestion
-- proved, then returns `Left errs`, where `errs` records all errors
-- invoked during proving.
```

(Mentioning the `[Error]` type is a bit redundant, given that it's already mentioned in the type signature.)